src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,51 @@
+import { useEffect } from 'react';
+
+import TodoItem from '../TodoItem/TodoItem';
+
+import { getTodos } from '../../api/todos';
+import { useTodos } from '../../hooks/useTodos';
+import { wait } from '../../utils/fetchClient';
+import { Filter } from '../../types';
+
+export const TodoList: React.FC = () => {
+  const { todos, setTodos, filter, setError, setIsLoading } = useTodos();
+
+  useEffect(() => {
+    const fetchTodos = async () => {
```

Comment:
Also, it's better to fetch todos in the parent component, and pass them to the todo list so it wouldn't know about any logic, just receive todos and render them